File: functions/src/types/weather.ts
// Weather-specific types and interfaces

export interface WeatherRequest {
  latitude: number;
  longitude: number;
  units?: "metric" | "imperial";
}

export interface ForecastRequest {
  latitude: number;
  longitude: number;
  units?: "metric" | "imperial";
}

export interface WeatherData {
  temperature: number;
  condition: string;
  humidity: number;
  windSpeed: number;
  windDirection: string;
  pressure: number;
  location: string;
  timestamp: string;
}

export interface ForecastDay {
  date: string;
  dayName: string;
  highTemp: number;
  lowTemp: number;
  condition: string;
  icon: string;
  humidity: number;
  windSpeed: number;
  windDirection: string;
  pressure: number;
  precipitation: number;
}

export interface ForecastData {
  location: string;
  days: ForecastDay[];
}

export interface WeatherResponse {
  success: boolean;
  data: WeatherData;
  cached: boolean;
  error?: string;
}

export interface ForecastResponse {
  success: boolean;
  data: ForecastData;
  cached: boolean;
  error?: string;
}

// OpenWeatherMap API response types
export interface OpenWeatherMain {
  temp: number;
  temp_min: number;
  temp_max: number;
  humidity: number;
  pressure: number;
}

export interface OpenWeatherWeather {
  description: string;
  icon: string;
}

export interface OpenWeatherWind {
  speed: number;
  deg: number;
}

export interface OpenWeatherCurrentResponse {
  main: OpenWeatherMain;
  weather: OpenWeatherWeather[];
  wind: OpenWeatherWind;
  name: string;
  sys: {
    country: string;
  };
}

export interface OpenWeatherForecastItem {
  dt: number;
  main: OpenWeatherMain;
  weather: OpenWeatherWeather[];
  wind: OpenWeatherWind;
  pop: number;
}

export interface OpenWeatherForecastResponse {
  city: {
    name: string;
    country: string;
    timezone: number;
  };
  list: OpenWeatherForecastItem[];
}

File: backend/internal/model/weather.go
package model

import "time"

type WeatherData struct {
	Location      string    `json:"location"`
	Temperature   float64   `json:"temperature"`
	Condition     string    `json:"condition"`
	Humidity      int       `json:"humidity"`
	WindSpeed     float64   `json:"wind_speed"`
	WindDirection string    `json:"wind_direction"`
	Pressure      float64   `json:"pressure"`
	UVIndex       int       `json:"uv_index"`
	FeelsLike     float64   `json:"feels_like"`
	Timestamp     time.Time `json:"timestamp"`
}

type ForecastDay struct {
	Date          string  `json:"date"`
	DayName       string  `json:"day_name"`
	HighTemp      float64 `json:"high_temp"`
	LowTemp       float64 `json:"low_temp"`
	Condition     string  `json:"condition"`
	Icon          string  `json:"icon"`
	Humidity      int     `json:"humidity"`
	WindSpeed     float64 `json:"wind_speed"`
	WindDirection string  `json:"wind_direction"`
	Pressure      float64 `json:"pressure"`
	Precipitation int     `json:"precipitation"`
}

type ForecastData struct {
	Location string        `json:"location"`
	Days     []ForecastDay `json:"days"`
}

type Recommendation struct {
	ID          string `json:"id"`
	Type        string `json:"type"` // weather, calendar, clothing, general
	Title       string `json:"title"`
	Description string `json:"description"`
	Priority    string `json:"priority"` // high, medium, low
	Action      string `json:"action"`
}

